src/python/gudhi/cubical_complex.pyx

```diff
@@ -145,6 +146,32 @@ cdef class CubicalComplex:
             persistence_result = self.pcohptr.get_persistence(homology_coeff_field, min_persistence)
         return persistence_result
 
+    def persistence_generators(self, homology_coeff_field=11, min_persistence=0, persistence_dim_max = False):
+        """This function returns the persistence of the simplicial complex.
+
+        :param homology_coeff_field: The homology coefficient field. Must be a
+            prime number. Default value is 11.
+        :type homology_coeff_field: int.
+        :param min_persistence: The minimum persistence value to take into
+            account (strictly greater than min_persistence). Default value is
+            0.0.
+            Sets min_persistence to -1.0 to see all values.
+        :type min_persistence: float.
+        :param persistence_dim_max: If true, the persistent homology for the
+            maximal dimension in the complex is computed. If false, it is
+            ignored. Default is false.
+        :type persistence_dim_max: bool
+        :returns: The persistence of the simplicial complex, together with the corresponding generators, i.e., the positive and negative top-dimensional cells.
+        :rtype:  list of pairs(dimension, pair(index of positive top-dimensional cell, index of negative top-dimensional cell))
```

Comment:
I can imagine why you might want the top-dimensional cells, but then you aren't exactly returning the cell that creates or kills the cycle. When we build a lower-star filtration in a simplicial complex, we get the simplices, not their vertex with maximal filtration. It may get confusing.

Reply:
Yes, you are right. I am creating this branch mainly to do optimization / backprop on persistence diagrams, so this is why I am only interested in the input data, i.e., the top-dimensional cells. But as you said, these are not quite the positive and negative simplices. I'll modify the doc.